lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -236,8 +236,13 @@ def create_configured_data_providers(lookup_invocation, parent_data_provider)
       when 'json', 'yaml'
         create_data_provider(backend, parent_data_provider, KEY_V3_DATA_HASH, "#{backend}_data", { KEY_DATADIR => datadir }, paths)
       else
-        # Custom backend. Hiera v3 must be installed and it must be made aware of the loaded config
-        require 'hiera/config'
+        # Custom backend. Hiera v3 must be installed, it's logger configured, and it must be made aware of the loaded config
+        require 'hiera'
+        if @config.include?(KEY_LOGGER)
+          Hiera.logger = @config[KEY_LOGGER].to_s
+        else
+          Hiera.logger = 'console'
```

Comment:
@thallgren I've tested this out on another vm, and puppet infrastructure configure succeeds with this patch. The default to the console logger seems like a change from HieraPuppet, where you would end up with the puppet logger (so we're getting a lot of debug output in the run). https://github.com/thallgren/puppet/blob/afad625267b5d8bff4f36b28d65625707df5ac98/lib/hiera_puppet.rb#L65

We can add a :logger: 'puppet' to the pe-installer-shim's hiera.yaml easily enough, but given that HieraPuppet is no longer in use (if I'm understanding these changes), it seems like this constitutes a behavior change for anyone using hiera?